runtime/queries/column_timeseries.go

```diff
@@ -263,27 +292,30 @@ func (q *ColumnTimeseries) createTimestampRollupReduction(
 		if err != nil {
 			return nil, err
 		}
+
 		defer rows.Close()
-		results := make([]*runtimev1.TimeSeriesValue, 0, (q.Pixels+1)*4)
-		for rows.Next() {
-			var ts time.Time
-			var count float64
-			err = rows.Scan(&ts, &count)
+		results, err := rowsToData(rows)
+		if err != nil {
+			return nil, err
+		}
+
+		tsv := make([]*runtimev1.TimeSeriesValue, 0, len(results))
+		for _, r := range results {
```

Comment:
Same point as before about creating `TimeSeriesValue` directly when iterating over `rows`. If `rowsToData` is used in other metrics view endpoints, we can add a new helper func `rowsToTimeSeriesValue` or something like that